src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2709,7 +2709,7 @@ class StubGenerator: public StubCodeGenerator {
     __ ctzc_bit(trailing_zeros, match_mask, haystack_isL, tmp, ch2);
     __ addi(trailing_zeros, trailing_zeros, haystack_isL ? 7 : 15);
     __ slli(needle_len, needle_len, BitsPerByte * wordSize / 2);
-    __ orr(haystack_len, haystack_len, needle_len); // restore needle_len(32bits)
+    __ orr(haystack_len, haystack_len, needle_len); // restore needle_len(3right_2_bits)
```

Comment:
Ah, this looks strange. Why replace `32bits` with `3right_2_bits` in your last commit?